app/models/user/reputation_acquisition.rb

```diff
@@ -2,7 +2,19 @@ class User::ReputationAcquisition < ApplicationRecord
   belongs_to :user
   belongs_to :reason_object, polymorphic: true, optional: true
 
+  before_create do
+    raise ReputationAcquisitionReasonMissingAmount unless self.amount || REASON_AMOUNTS[self.reason]
```

Comment:
Rather than raising in `before_create` use validators.

Reply:
Also, I'd probably go for `REASON_AMOUNTS.has_key?(self.reason)`

Reply:
And I think that we should always check the reason is valid, not only if the `amount` is missing

Reply:
I've converted to `validates`, which fixes the above comments, right?